aws-vault.go

```diff
@@ -21,44 +19,27 @@ func main() {
 		ErrorWriter: os.Stderr,
 	}
 
-	// handle profile at the top level, I always do this.
```

Comment:
This might be annoying, but I think it's kind of horrible having it here as it introduces a leaky abstraction and strange behaviour like needing to support the profile flag in both spots.